CMakeLists.txt

```diff
@@ -397,6 +397,7 @@ link_libraries(global-group)
 target_link_libraries(global-group INTERFACE $<TARGET_PROPERTY:global-libs,INTERFACE_LINK_LIBRARIES>)
 
 option (ENABLE_LIBFIU "Enable libfiu" ON)
+option (FIU_ENABLE "Enable fiu" ON)
```

Comment:
If I interpret the original problem correctly, then the problem is that disabling `ENABLE_LIBFIU` at CMake configure time breaks the universe (compilation).

The problem is that adding another CMake options is too confusing. We should really keep only the existing only but make it work properly (build libfiu - but disable `FIU_ENABLE`).

(I will also say that such kind of behavior is still kind of unconventional in ClickHouse. Normally, disabling `ENABLE_LIBXYZ` means that a library is truly not compiled. So we should at least add comments here to explain what is happening.)